src/main/scala/notifications/kafka/Producer.scala

```diff
@@ -0,0 +1,37 @@
+package notifications.kafka
+
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+import providers.WeatherClient
+import providers.openweathermap.Responses.Weather
+import server.database.DatabaseInterface
+
+class Producer(mariaDb: DatabaseInterface, client: WeatherClient, config: WeatherProducerConfig) extends Runnable {
+  val producer = new KafkaProducer[String, Weather](config.props)
+
+  def getActiveLocations(): Seq[String] = mariaDb.getLocationsWithActiveAlerts().map(_.value)
+
+  def getLocationData(location: String): Weather = {
+    client.getWeatherData(Seq(("q", location)))
+  }
+
+  def produceLocationData(location: String) = {
+    val weather = getLocationData(location)
+    val data = new ProducerRecord[String, Weather](config.topic, location, weather)
+    producer.send(data)
+  }
+
+  override def run(): Unit = {
+    try {
+      while (true) {
```

Comment:
I suppose there is no scala alternative to doing this?

Reply:
you mean using something else than Threads? should we maybe run them as separate jars?

Reply:
I meant not using while(true) and try {}. I hardly ever see that used in scala